marklogic-client-api-functionaltests/src/test/java/com/marklogic/client/fastfunctest/TestSearchPageLength.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright © 2024 MarkLogic Corporation. All Rights Reserved.
+ */
+
+package com.marklogic.client.fastfunctest;
+
+import com.marklogic.client.impl.OkHttpServices;
+import com.marklogic.client.impl.RESTServices;
+import com.marklogic.client.io.*;
+import com.marklogic.client.query.*;
+import org.custommonkey.xmlunit.exceptions.XpathException;
+import org.junit.jupiter.api.*;
+import org.w3c.dom.Document;
+
+import static org.custommonkey.xmlunit.XMLAssert.*;
+
+
+class TestSearchPageLength extends AbstractFunctionalTest {
+
+	static final QueryManager queryMgr = client.newQueryManager();
+	StringQueryDefinition qd = queryMgr.newStringDefinition();
+
+	@Test
+	void testSearchPageLengthZero() throws XpathException {
+		queryMgr.setPageLength(0);
+		qd.setCriteria("Foo");
+		DOMHandle resultsHandle = new DOMHandle();
+		queryMgr.search(qd, resultsHandle);
+
+		Document resultDoc = resultsHandle.get();
+		assertXpathEvaluatesTo("0", "string(//*[local-name()='response']//@*[local-name()='page-length'])", resultDoc);
+	}
+
+
+	@Test
+	void testSearchPageLengthNegative() throws XpathException {
+		queryMgr.setPageLength(-1);
+		qd.setCriteria("Foo");
+		DOMHandle resultsHandle = new DOMHandle();
+		queryMgr.search(qd, resultsHandle);
+
+		Document resultDoc = resultsHandle.get();
+		assertXpathEvaluatesTo("10", "string(//*[local-name()='response']//@*[local-name()='page-length'])", resultDoc);
```

Comment:
This is surprising behavior by the server - i.e. it appears to be ignoring a negative value and using the server default. That is worth capturing in an assertion message - i.e. that we don't expect an error from the server, the server just ignores the input.

Reply:
The server isn't ignoring the negative value, it isn't getting passed.
Why would you have a negative page length?

Reply:
Oh right - that's what the assertion message should state then - i.e. that a negative value isn't sent (instead of an error being thrown), so we expect the server default page length of 10 to be used.